patches/tModLoader/Terraria/release_extras/LaunchUtils/ScriptCaller.sh

```diff
@@ -6,7 +6,13 @@
 cd "$(dirname "$0")"
 . ./BashUtils.sh
 
-echo "You are on platform: \"$_uname\""
+echo "You are on platform: \"$_uname\" arch: \"$_arch\""
+
+# Detect arm64 launches and the presence of rosetta (stolen from the dotnet official install script)
+if [ "$_arch" = "arm64" ] && [ "$(/usr/bin/pgrep oahd >/dev/null 2>&1;echo $?)" -eq 0 ]; then
+	echo "arm64 detected, restarting under arch -x86_64"
+	exec arch -x86_64 ./ScriptCaller.sh "$@"
```

Comment:
The second calculation is hard to tell at a glance that it limits to Mac's specifically & does not impact Linux Arm64. 
Please add a comment about that the latter 'checks if the Rosetta process is running'.

Not clear at a glance if that check is system-wide or session specific, likely worth a confirmation test of have an app running on rosetta than launch this & verify this does trigger

Reply:
I did add that comment, explicitly, on the line above. `arch` changes it for the process tree of the called command.

Reply:
Apologies, I'll clarify - 
The `/usr/bin/pgrep oahd >/dev/null 2>&1` checks if it is running, not that it is installed.  Good to clarify 'presence'. Did miss the rosetta portion of that comment and only somehow saw arm64 and part in brackets. Apologies.

My concern is that it is 'running' for a second process already, So the order of ops would go:
1) App 1 is running with Rosetta in background
2) Launch tML
3) `/usr/bin/pgrep oahd >/dev/null 2>&1` returns '1' since 'Rosetta is running', even if not in the current terminal context of first launch
4) And thus it continues to run past that point as if it is non-Rosetta Arm64.

Reply:
`oahd` is a system process which indicates that Rosetta is present on the machine. At least, that's my understanding. The bash script will not be 'running under rosetta' at all. Only when an x86 process is launched is Rosetta 'running' in a sense. Using `arch -x86_64` changes the environment for the process tree so that the dotnet download script can detect the right architecture.

Reply:
To paraphase back your intended design:
```# Detect arm64 arch and the presence of rosetta via `oahd`, and if present, restart with forced usage of x86 dotnet runtime```

No further concerns with approach, misunderstood as attempting to force rosetta in the terminal itself.